File: src/main/java/no/nav/tjenestepensjon/simulering/consumer/FindTpLeverandorCallable.java
package no.nav.tjenestepensjon.simulering.consumer;

import java.util.List;
import java.util.concurrent.Callable;

import no.nav.tjenestepensjon.simulering.domain.TPOrdning;
import no.nav.tjenestepensjon.simulering.domain.TpLeverandor;

public class FindTpLeverandorCallable implements Callable<TpLeverandor> {

    private final TPOrdning tpOrdning;
    private final TpConfigConsumer tpConfigConsumer;
    private final List<TpLeverandor> tpLeverandorList;

    public FindTpLeverandorCallable(TPOrdning tpOrdning, TpConfigConsumer tpConfigConsumer, List<TpLeverandor> tpLeverandorList) {
        this.tpOrdning = tpOrdning;
        this.tpConfigConsumer = tpConfigConsumer;
        this.tpLeverandorList = tpLeverandorList;
    }

    @Override
    public TpLeverandor call() throws Exception {
        String tpLeverandor = tpConfigConsumer.findTpLeverandor(tpOrdning);
        return tpLeverandorList.stream().filter(l -> tpLeverandor.equalsIgnoreCase(l.getName())).findFirst().get();
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/consumer/FindTpLeverandorCallable.kt
package no.nav.tjenestepensjon.simulering.consumer

import no.nav.tjenestepensjon.simulering.model.v1.domain.TPOrdning
import no.nav.tjenestepensjon.simulering.domain.TpLeverandor
import java.util.concurrent.Callable

class FindTpLeverandorCallable(private val tpOrdning: TPOrdning, private val tpConfigConsumer: TpConfigConsumer, private val tpLeverandorList: List<TpLeverandor>) : Callable<TpLeverandor> {
    @Throws(Exception::class)
    override fun call(): TpLeverandor {
        val tpLeverandor = tpConfigConsumer.findTpLeverandor(tpOrdning)
        return tpLeverandorList.first { l: TpLeverandor -> tpLeverandor.equals(l.name, ignoreCase = true) }
    }

}